test/acceptance/helpers/helpers.go

```diff
@@ -148,3 +152,95 @@ func Cleanup(t *testing.T, noCleanupOnFailure bool, cleanup func()) {
 
 	t.Cleanup(wrappedCleanupFunc)
 }
+
+// WritePodsDebugInfoIfFailed calls kubectl describe and kubectl logs --all-containers
+// on pods filtered by the labelSelector and writes it to the debugDirectory.
+func WritePodsDebugInfoIfFailed(t *testing.T, kubectlOptions *k8s.KubectlOptions, debugDirectory, labelSelector string) {
+	t.Helper()
+
+	if t.Failed() {
+		// Create k8s client from kubectl options
+		client := KubernetesClientFromOptions(t, kubectlOptions)
+
+		contextName := kubernetesContextFromOptions(t, kubectlOptions)
+
+		// Create a temporary directory if one isn't provided.
+		if debugDirectory == "" {
+			var err error
+			debugDirectory, err = ioutil.TempDir("", "consul-test")
```

Comment:
Not a blocker by any means, but I can imagine that this will overwrite the last test's results, and maybe it would be helpful to be able to compare the two while debugging something?
Would it make sense to always use `consul-test` or should we salt the name with some sort of identifier so we don't overwrite the previous test results?

Reply:
Actually it looks like you're doing that when generating testDebugDirectory() .. nevermind! 👍🏽